src/main/java/net/revelc/code/formatter/ValidateMojo.java

```diff
@@ -56,11 +77,24 @@ protected void doFormatFile(final File file, final ResultCollector rc, final Pro
         super.doFormatFile(file, rc, hashCache, basedirPath, true);
 
         if (rc.successCount != 0) {
-            throw new MojoFailureException("File '" + file
-                    + "' has not been previously formatted.  Please format file and commit before running validation!");
+            String errorMessage = String.format(
+                    "File '%s' has not been previously formatted. Please format file (for example by invoking `%s`) and commit before running validation!",
+                    file, formatCommand());
+            throw new MojoFailureException(errorMessage);
         }
         if (rc.failCount != 0) {
-            throw new MojoExecutionException("Error formating '" + file + "' ");
+            throw new MojoExecutionException("Error formatting '" + file + "' ");
+        }
+    }
+
+    private String formatCommand() {
+        String validateInvocation = String.format("mvn %s:%s:%s:%s@%s", mojoGroupId, mojoArtifactId, mojoVersion,
+                FORMAT_MOJO_NAME, mojoExecution.getExecutionId());
```

Comment:
I don't think you'd want to include the execution ID here. What purpose does that serve?

Reply:
As @famod mentions above, it's used to ensure that setups that contain multiple executions of the plugin will have the proper ID.

Reply:
My thinking was that this would only execute using the config that was set up for verification. I'm not sure that makes sense. What if they have a config specifically for formatting? Although executing with the same config that is causing validate to fail is more likely to get the validation to succeed, it may not be the best command to execute. It also wouldn't work if the execution ID is defined inside a profile, unless the user also activated that profile as well.

Since we're merely making a suggestion, it's probably better to just omit the execution ID, and leave it up to the user. That way, we don't risk giving bad advice.

Also, since presumably the version is already specified inside this plugin's POM, we can probably omit the version as well, to streamline the suggestion.

Reply:
To be clear, this isn't really an objection. Just discussing. If others think it's good to have the version and the execution ID, I'm fine with it.

Reply:
In the end it's your choice @ctubbsii but when specifying the execution id you will always end up with the properly merged config of all "sources", e.g. including profiles and plugin-wide config.
AFAICS this also includes the execution id. What the plugin sees via `MojoExecution` should always be the "effective model".

Reply:
The execution ID can be reused by multiple execution definitions, resulting in effective models in the end.

The most common is the how users often clobber the default execution ID of a plugin that is bound to the default build lifecycle to change its behavior or to prevent it from running (by giving it a fake phase), as in https://stackoverflow.com/questions/34590452/what-is-the-implicit-id-of-a-maven-plugin-execution

You can also clobber an execution ID and change the effective model for that execution by reusing an execution ID from an execution defined in a parent POM.

But, the one I'm thinking about is when an execution ID is reused inside a profile, in order to conditionally clobber an execution defined outside the profile (or in a different profile). In order to get the same effective model for the suggested format command, you'd also have to activate the same set of profiles that were active when the verification failed.

For example, if you run `mvn -Prelease formatter:validate` and it fails, these changes might suggest running `mvn net.revelc.code.formatter:formatter-maven-plugin:2.17.1:format@validateReleaseFormatting`, even though that won't work without also adding `-Prelease` if the formatter execution ID was defined inside the release profile.

Reply:
> you'd also have to activate the same set of profiles that were active when the verification failed

Yeah, totally. That's what I meant earlier today, but it makes the command even longer.

I think the bottom line for now is to keep it simple(r) and see how it goes.
I'm fine with dropping the execution id.

Reply:
Okay, removed